### PR TITLE
Unpin TensorFlow

### LINKED ISSUE #1819: TF 2.3.0 incompatibility
TF 2.3.0 was released yesterday, and seems to be incompatible with TFP <0.11.0 and breaks imports (https://travis-ci.com/github/rlworkgroup/garage/jobs/365922927#L3061). We pin TFP to <=0.10.0 in the first place to resolve cloudpickle version mismatch (https://github.com/rlworkgroup/garage/issues/1758). Since TFP 0.11.0 still pins cloudpickle to 1.3.0 while gym needs 1.2.x, unpinning TFP won't work. So for now, TF needs to be pinned to <2.3.0

### PR DESCRIPTION
**Pull request by yeukfu**
This should close #1819.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,30 +4,27 @@
...
     'cma==2.7.0',
     'dowel==0.0.3',
     f'gym[atari,box2d,classic_control]=={GYM_VERSION}',
     'numpy>=1.14.5',
     'psutil',
-    # Pyglet 1.4.0 introduces some api change which breaks some
-    # gym environments
-    # See: https://github.com/openai/gym/issues/1588
-    'pyglet<1.4.0,>=1.3.0',
+    'pyglet>=1.4.0,<=1.5.0',
```
Comment (gitanshu) [inline setup.py:20]: I think we can remove this as a direct dependency if we are upgrading gym

File: setup.py [hunk 2]
```diff
@@ -4,30 +4,27 @@
...
-GYM_VERSION = '0.15.4'
+GYM_VERSION = '0.17.2'
 
 # Required dependencies
 REQUIRED = [
     # Please keep alphabetized
     'akro',
     'click>=2.0',
-    'cloudpickle<1.5',
+    'cloudpickle==1.3',
```
Comment (gitanshu) [inline setup.py:14]: Can we do `>=1.3` instead? This seems too restrictive

File: setup.py (hunk 1 above)
Comment (yeukfu) [reply setup.py:20]: I copy this form gym's [setup.py](https://github.com/openai/gym/blob/6df1b994bae791667a556e193d2a215b8a1e397a/setup.py#L31)

File: setup.py (hunk 2 above)
Comment (yeukfu) [reply setup.py:14]: It seems that tensorflow-probability 0.11 pin cloudpickle to 1.3.

Comment (gitanshu) [reply setup.py:14]: Ahh right. It's set to >=1.3 on master, but that's unreleased yet. Can you open an issue to track this?

Comment (yeukfu) [reply setup.py:14]: Sure. So currently should we pin cloudpickle to 1.3 or >=1.3?

Comment (gitanshu) [reply setup.py:14]: I guesswe can leave it at ==1.3 for now

**Comment by ryanjulian** [comment]
@Mergifyio rebase

**Comment by yeukfu** [comment]
@Mergifyio backport release-2020.06

**Pull request merged**